packages/@aws-cdk/aws-codedeploy/lib/server/load-balancer.ts

```diff
@@ -0,0 +1,90 @@
+import elb = require('@aws-cdk/aws-elasticloadbalancing');
+import elbv2 = require('@aws-cdk/aws-elasticloadbalancingv2');
+
+/**
+ * The generations of AWS load balancing solutions.
+ */
+export enum LoadBalancerGeneration {
+  /**
+   * The first generation (ELB Classic).
+   */
+  FIRST = 0,
+
+  /**
+   * The second generation (ALB and NLB).
+   */
+  SECOND = 1
+}
+
+/**
+ * An interface of an abstract load balancer, as needed by CodeDeploy.
+ * Create instances using the static factory methods:
+ * {@link #classic}, {@link #alb} and {@link #nlb}.
+ */
+export abstract class LoadBalancer {
+  /**
+   * Creates a new CodeDeploy load balancer from a Classic ELB Load Balancer.
+   *
+   * @param loadBalancer a classic ELB Load Balancer
+   */
+  public static classic(loadBalancer: elb.LoadBalancer): LoadBalancer {
+    return new ClassicLoadBalancer(loadBalancer);
+  }
+
+  /**
+   * Creates a new CodeDeploy load balancer from an Application Load Balancer Target Group.
+   *
+   * @param albTargetGroup an ALB Target Group
+   */
+  public static alb(albTargetGroup: elbv2.ApplicationTargetGroup): LoadBalancer {
+    return new AlbLoadBalancer(albTargetGroup);
+  }
+
+  /**
+   * Creates a new CodeDeploy load balancer from a Network Load Balancer Target Group.
+   *
+   * @param nlbTargetGroup an NLB Target Group
+   */
+  public static nlb(nlbTargetGroup: elbv2.NetworkTargetGroup): LoadBalancer {
+    return new NlbLoadBalancer(nlbTargetGroup);
+  }
+
+  public abstract readonly generation: LoadBalancerGeneration;
+  public abstract readonly name: string;
+}
+
+class ClassicLoadBalancer extends LoadBalancer {
```

Comment:
You could implement this as inline classes, will be much more concise

Reply:
Done.